src/machine/machine_nxpmk66f18.go

```diff
@@ -83,3 +109,147 @@ func (p FastPin) Clear()       { p.PCOR.Set(true) }
 func (p FastPin) Toggle()      { p.PTOR.Set(true) }
 func (p FastPin) Write(v bool) { p.PDOR.Set(v) }
 func (p FastPin) Read() bool   { return p.PDIR.Get() }
+
+// Configure the UART.
+func (u UART) Configure(config UARTConfig) {
+	if u == USBUART {
+		return
+	} else if u == LPUART {
+		return
+	}
+	r := u.reg()
+	s := uartState[u]
+
+	// turn on the clock
+	r.SCGC.Set(r.SCGCMask)
+
+	// configure pins
+	r.RXPCR.Set(nxp.PORT_PCR0_PE | nxp.PORT_PCR0_PS | nxp.PORT_PCR0_PFE | nxp.PORT_PCR0_MUX(3))
+	r.TXPCR.Set(nxp.PORT_PCR0_DSE | nxp.PORT_PCR0_SRE | nxp.PORT_PCR0_MUX(3))
+
+	// default to 115200 baud
+	if config.BaudRate == 0 {
+		config.BaudRate = 115200
+	}
+
+	// copied from teensy core's BAUD2DIV macro
+	divisor := ((CPUFrequency() * 2) + ((config.BaudRate) >> 1)) / config.BaudRate
+
+	r.BDH.Set(uint8((divisor >> 13) & 0x1F))
+	r.BDL.Set(uint8((divisor >> 5) & 0xFF))
+	r.C4.Set(uint8(divisor & 0x1F))
+	r.C1.Set(nxp.UART_C1_ILT)
+	r.TWFIFO.Set(2) // tx watermark, causes S1_TDRE to set
+	r.RWFIFO.Set(4) // rx watermark, causes S1_RDRF to set
+	r.PFIFO.Set(nxp.UART_PFIFO_TXFE | nxp.UART_PFIFO_RXFE)
+	r.C2.Set(uartC2TXInactive)
+
+	// if s.Interrupt == interrupt.Interrupt{} {
+	s.Interrupt = u.newISR()
+	s.Interrupt.SetPriority(uartIRQPriority)
+	s.Interrupt.Enable()
```

Comment:
I don't think you'll need an interrupt for sending bytes.

Reply:
I was translating Paul's code. In general, I'm trying to avoid reimplementing as much as I can. Understanding how to write low-level K66 code is not my goal.